bindings/python/py_src/safetensors/torch.py

```diff
@@ -36,7 +36,7 @@ def storage_size(tensor: torch.Tensor) -> int:
 def _find_shared_tensors(state_dict: Dict[str, torch.Tensor]) -> List[Set[str]]:
     tensors = defaultdict(set)
     for k, v in state_dict.items():
-        if v.device != torch.device("meta"):
+        if v.device != torch.device("meta") and storage_ptr(v) != 0 and storage_size(v) != 0:
```

Comment:
Because it's the storage pointer, not the tensor I'm looking for.
You can have an empty tensor, that still is backed by shared storage, in which case I still want to yell.

Reply:
Hum so why not do `storage_size(v) != 0` only? You should accept any tensor that has empty storage regardless on where they are actually stored no?

Reply:
No, I'm also trying to safeguard users from doing wrong things.
Saving empty tensors should worry most users.

Here this was never allocated in the first place, so it looks intentional.

Reply:
I don't understand. Am I right in the following:
```python
torch.empty([]) # want to supported
torch.zeros((2,0)) # want to support
torch.zeros((2,1))[:, :0] # no support
```
If so, `storage_size(v) != 0` should work.

Reply:
```python
>>> l = [torch.tensor([]), torch.zeros((2,0)), torch.zeros((2,1))[:, :0]]
>>> for elt in l:
...   print(elt.untyped_storage().size())
... 
0
0
8
```

Reply:
`torch.zeros((2,1))[:, :0]` this will work if it's alone in the state dict actually. But if 2 tensors share the same storage, then crash will happen. I don't really want to start looking if the slices actually overlap or not.

Reply:
Hum what I mean is if your storage is 0, there's no overlap or anything.

Reply:
`torch.zeros((2,1))[:, :0]` has not a `0` sized storage.